DSharpPlus.Commands/Processors/SlashCommands/SlashCommandProcessor.cs

```diff
@@ -0,0 +1,442 @@
+            allowDMUsage: command.Attributes.Any(x => x is AllowDMUsageAttribute),
+            defaultMemberPermissions: command.Attributes.OfType<RequirePermissionsAttribute>().FirstOrDefault()?.UserPermissions ?? Permissions.None,
+            nsfw: command.Attributes.Any(x => x is RequireNsfwAttribute)
+        );
+    }
+
+    [SuppressMessage("Roslyn", "CA1859", Justification = "Incorrect warning in NET 8-rc2. TODO: Open an issue in dotnet/runtime.")]
+    public async Task<DiscordApplicationCommandOption> ToApplicationParameterAsync(Command command)
+    {
+        if (this._extension is null)
+        {
+            throw new InvalidOperationException("SlashCommandProcessor has not been configured.");
+        }
+
+        // Convert the subcommands or parameters into application options
+        List<DiscordApplicationCommandOption> options = [];
+        if (command.Subcommands.Any())
+        {
+            foreach (Command subCommand in command.Subcommands)
+            {
+                options.Add(await this.ToApplicationParameterAsync(subCommand));
+            }
+        }
+        else
+        {
+            foreach (CommandParameter parameter in command.Parameters)
+            {
+                options.Add(await this.ToApplicationParameterAsync(command, parameter));
+            }
+        }
+
+        // Translate the subcommand's name and description.
+        IReadOnlyDictionary<string, string> nameLocalizations = new Dictionary<string, string>();
+        IReadOnlyDictionary<string, string> descriptionLocalizations = new Dictionary<string, string>();
+        if (command.Attributes.OfType<SlashLocalizerAttribute>().FirstOrDefault() is SlashLocalizerAttribute localizerAttribute)
+        {
+            AsyncServiceScope scope = this._extension.ServiceProvider.CreateAsyncScope();
+            nameLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.name");
+            descriptionLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.description");
+            await scope.DisposeAsync();
+        }
+
+        return new(
+            name: nameLocalizations.TryGetValue("en-US", out string? name) ? name : command.Name,
+            description: descriptionLocalizations.TryGetValue("en-US", out string? description) ? description : command.Description,
+            name_localizations: nameLocalizations,
+            description_localizations: descriptionLocalizations,
+            type: command.Subcommands.Any() ? ApplicationCommandOptionType.SubCommandGroup : ApplicationCommandOptionType.SubCommand,
+            options: options
+        );
+    }
+
+    [SuppressMessage("Roslyn", "CA1859", Justification = "Incorrect warning in NET 8-rc2. TODO: Open an issue in dotnet/runtime.")]
+    private async Task<DiscordApplicationCommandOption> ToApplicationParameterAsync(Command command, CommandParameter parameter, int? i = null)
```

Comment:
Maybe a bit more describtive parameter name.

Reply:
It's... the parameter you're converting to an application parameter. How much more descriptive could it get?

Reply:
I was talking about the last parameter, its confusing when you read a long method and then theres a random `i`

Reply:
OH. Yeah I can do that